src/main/java/domain/LottoResult.java

```diff
@@ -0,0 +1,139 @@
+package domain;
+
+import constants.LottoWinningPrize;
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class LottoResult {
+
+    private int threeLottoWinning;
+    private int fourLottoWinning;
+    private int fiveLottoWinning;
+    private int fiveBonusLottoWinning;
+    private int sixLottoWinning;
+    private boolean isBonusLotto;
+
+    private BigDecimal totalProfit;
+    private BigDecimal profitRate;
+
+
+    public int getEqualsLottoCount(List<Integer> lottoNumbers, UserLotto userLotto){
+        Set<Integer> lottoSet = new HashSet<>(lottoNumbers);
+        int winningCount= 0;
+
+        for (Integer userNumber : userLotto.getLottoNumbers()) {
+            if (lottoSet.contains(userNumber)) {
+                winningCount++;
+            }
+        }
+
+        if(lottoSet.contains(userLotto.getBonusNumber())){
+            this.isBonusLotto = true;
+            winningCount++;
+        }
+
+        return winningCount;
```

Comment:
보너스번호가 포함되어져있으면 winningCount가 증가하게 되는데 보너스번호는 6개중에 5개를 맞춘 후 2등인지 3등인지 판단을 할 때만 검사를 해야됩니다 현재 해당코드는 정답번호와 맞은개수가 상관없이 보너스 번호가 맞으면 정답번호와 동일하게 판단이 되서 오류가 발생합니다 ex) 정답번호 5개 보너스번호 1개가 맞은경우 1등이됨
![image](https://github.com/woowacourse-precourse/java-lotto-6/assets/127712161/5e7b7351-9b23-410e-88b8-7bfb3777e219)

Reply:
> 보너스번호가 포함되어져있으면 winningCount가 증가하게 되는데 보너스번호는 6개중에 5개를 맞춘 후 2등인지 3등인지 판단을 할 때만 검사를 해야됩니다 현재 해당코드는 정답번호와 맞은개수가 상관없이 보너스 번호가 맞으면 정답번호와 동일하게 판단이 되서 오류가 발생합니다 ex) 정답번호 5개 보너스번호 1개가 맞은경우 1등이됨 ![image](https://user-images.githubusercontent.com/127712161/282378396-5e7b7351-9b23-410e-88b8-7bfb3777e219.png)

헉...... 그러게요 이렇게 코드를 수정해야겠네요
저도 테스트하면서 발견하지 못했는데 꼼꼼히 확인 하시고 오류 발견 해 주셔서 감사합니다!

```suggestion
        if(lottoSet.contains(userLotto.getBonusNumber())){
            this.isBonusLotto = true;
        }
```